dc/s2n-quic-dc/src/stream/send/state.rs

```diff
@@ -817,9 +813,14 @@ impl State {
 
     #[inline]
     fn force_arm_pto_timer(&mut self, clock: &impl Clock) {
-        let pto_period = self
+        let mut pto_period = self
             .rtt_estimator
             .pto_period(self.pto_backoff, PacketNumberSpace::Initial);
+
+        // the `Timestamp::elapsed` function rounds up to the nearest 1ms so we need to set a min value
+        // otherwise we'll prematurely trigger a PTO
+        pto_period = pto_period.max(Duration::from_millis(2));
```

Comment:
Was this the source of the bug or did both changes contribute?

Reply:
The other change was the main source of reduction. Without the rounding change we would transmit an occasional PTO before the peer even had a chance to respond with an ACK.